modules/aioseop_sitemap.php

```diff
@@ -2204,9 +2204,35 @@ public function get_sitemap_index_filenames() {
 
 			$files = apply_filters( 'aioseop_sitemap_index_filenames', $files, $prefix, $suffix );
 
+			// Remove Additional Pages index if all pages are static and no extra pages are specified.
+			if ( ! $this->does_addl_sitemap_contain_urls() ) {
+				$page_to_remove = array( get_site_url() . '/addl-sitemap.xml' );
+				$files = $this->remove_urls_from_sitemap_page( $files, $page_to_remove );
+			}
+
 			return $files;
 		}
 
+		/**
+		 * The does_addl_sitemap_contain_urls() function.
+		 *
+		 * Checks whether the Additional Pages index will contain URLs.
+		 * This will not be the case if there is both a static homepage/posts page and there are no additional pages specified.
+		 *
+		 * @since 3.2.0
+		 *
+		 * @return bool
+		 */
+		private function does_addl_sitemap_contain_urls() {
+			$is_addl_pages = $this->option_isset( 'aiosp_sitemap_addl_pages' );
+			if ( ! $is_addl_pages &&
+				( 0 !== get_option( 'page_on_front' ) ) &&
```

Comment:
Should this be an `||`? Just in case one static page setting is used and not the other.

Reply:
No, this function is meant to check whether the addl sitemap file should be included in the root sitemap file. If there is no static homepage AND no static posts page AND no additional pages specified in the XML Sitemap menu, then there should be no addl sitemap file. If either of those conditions is true then it should be included.